### ISSUE TITLE
R.swift causing .XLIFF export to have no comment?

### ISSUE THREAD

**Issue by pschneider** [mac-cain13/R.swift#342]
Hi,

we use the .XLIFF Export Mechanism provided by Xcode to translate our app. We recently added a new localization to our project `de_AT`. Our base is `en` and we also support `de` already. Exporting the translations again seems to remove a lot of `<note>`-Tags from the existing .XLIFF file. 

For example:
Removes
```
<note>Summary label singular in checkout</note>
```
Adds
```
<note>No comment provided by engineer.</note>
```

But both `Localizable.strings` - files are looking like this (its the same translation for de or de_AT):
```
/* Summary label singular in checkout */
"%d Article - Sum" = "%d Artikel - Summe";
```

The code calling this looks like this:

```
String(format: NSLocalizedString("%d Article - Sum", comment: "Summary label singular in checkout"), 1)
```

So as far as I can tell, the `<note>` in the exported XLIFF file should not be `No comment provided by engineer.`, right?

Searching for the string finds the following `R.generated` entry and I think this entry may be causing the `note` tag to be reset:

```/// de translation: %d Artikel - Summe
      /// 
      /// Locales: de, de-AT
      static func dArticleSum(_ value1: Int) -> String {
        return String(format: NSLocalizedString("%d Article - Sum", bundle: R.hostingBundle, comment: ""), locale: R.applicationLocale, value1)
      }
```

Is it possible that Xcode overwrites this because it also looks in the `R.generated` file for `NSLocalizedString` usage and this contains no comment? Why are comments missing here?

Any help appreciated, thanks!

**Comment by tomlokhorst** [comment]
Xcode xliff export does indeed merge .strings files with occurrences of `NSLocalizedString` in code. My guess is that this is caused by the `comment: ""` in the generated code.

This may be related to https://github.com/mac-cain13/R.swift/issues/232

I will look into this further.

**Comment by mykola-cookpad** [comment]
@tomlokhorst have you had chance to look into this? it's still a problem

**Comment by tomlokhorst** [comment]
Should be fixed with version 7.0.0, since no calls to `NSLocalizedString` are generated anymore.
